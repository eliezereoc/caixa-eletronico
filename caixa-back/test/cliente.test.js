const request = require("supertest");
const server = require("../src/server/server");

const cliente = {
  id: "1",
  nome: "Eliezer de Oliveira Cardoso",
  saldo: "5000",
  saque: "0",
};

const updateCliente = {
  id: "1",
  nome: "Eliezer de Oliveira",
  saldo: "10000",
  saque: "0",
};

describe("Teste de Integração - Cliente", () => {
  test("Resposta status 200 na raiz", async () => {
    const res = await request(server).get("/");
    expect(res.status).toBe(200);
  });

  test("DELETE Cliente", async () => {
    const res = await request(server).delete("/cliente/1");

    expect(res.text).toBe("Cadastro excluido com sucesso!");
    expect(res.status).toBe(200);
  });

  test("POST Cliente", async () => {
    const res = await request(server).post("/cliente").send(cliente);
    expect(res.body.acknowledged).toBe(true);
    expect(res.status).toBe(200);
  });

  test("GET Cliente", async () => {
    const res = await request(server).get("/cliente").send(cliente);

    expect(res.body[0].id).toBe(cliente.id);
    expect(res.body[0].nome).toBe(cliente.nome);
    expect(res.body[0].saldo).toBe(cliente.saldo);
    expect(res.body[0].saque).toBe(cliente.saque);
    expect(res.status).toBe(200);
  });

  test("PUT Cliente", async () => {
    const res = await request(server).put("/cliente").send(updateCliente);

    expect(res.body.acknowledged).toBe(true);
    expect(res.status).toBe(200);
  });

  test("GET update Cliente", async () => {
    const res = await request(server).get("/cliente").send(updateCliente);
    expect(res.body[0].id).toBe(updateCliente.id);
    expect(res.body[0].nome).toBe(updateCliente.nome);
    expect(res.body[0].saldo).toBe(updateCliente.saldo);
    expect(res.body[0].saque).toBe(updateCliente.saque);
    expect(res.status).toBe(200);
  });
});
