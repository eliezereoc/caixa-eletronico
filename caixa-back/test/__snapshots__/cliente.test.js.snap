// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Integração DELETE Cliente: Cadastro excluido com sucesso! 1`] = `200`;

exports[`Teste de Integração GET Cliente 1`] = `Array []`;

exports[`Teste de Integração GET Cliente: 0 1`] = `undefined`;

exports[`Teste de Integração GET Cliente: 0 2`] = `undefined`;

exports[`Teste de Integração GET Cliente: 1 1`] = `
Object {
  "_id": "627013da3d609717287007de",
  "id": "1",
  "nome": "Eliezer de Oliveira Cardoso",
  "saldo": 10000,
  "saque": 0,
}
`;

exports[`Teste de Integração GET Cliente: 1 2`] = `
Object {
  "_id": "627013da3d609717287007de",
  "id": "1",
  "nome": "Eliezer de Oliveira Cardoso",
  "saldo": 10000,
  "saque": 0,
}
`;

exports[`Teste de Integração GET Cliente: 10000 1`] = `undefined`;

exports[`Teste de Integração GET Cliente: 10000 2`] = `undefined`;

exports[`Teste de Integração GET Cliente: Eliezer de Oliveira Cardoso 1`] = `undefined`;

exports[`Teste de Integração GET Cliente: Eliezer de Oliveira Cardoso 2`] = `undefined`;

exports[`Teste de Integração POST Cliente: 0 1`] = `undefined`;

exports[`Teste de Integração POST Cliente: 1 1`] = `undefined`;

exports[`Teste de Integração POST Cliente: 10000 1`] = `undefined`;

exports[`Teste de Integração POST Cliente: Eliezer de Oliveira Cardoso 1`] = `undefined`;
